### PR TITLE
Fix GitHubClassroom.github_url when using Enterprise

### PR DESCRIPTION
**Pull request by favonia**
Sorry that I do not know how to create a regression test, but this is definitely a bug to fix.

### CODE REVIEW

File: config/initializers/github_client.rb [hunk 1]
```diff
@@ -24,7 +24,7 @@ def self.enterprise?
 
   def self.github_url
     if enterprise?
```
Comment (stephaniegiang) [inline config/initializers/github_client.rb]: Hi @favonia 👋Thank you so much for contributing!

Looking at this PR, I think what you are trying to do is this: 

```suggestion
     if enterprise? && Rails.application.secrets.github_enterprise_url.present?
```

Comment (favonia) [reply config/initializers/github_client.rb]: I do not think this is necessary. `enterprise?` returns `true` only when `Rails.application.secrets.github_enterprise_url.present?` return trues. Please tell me if I was misguided.

https://github.com/education/classroom/blob/6561b4630e45e5455fdda6f479cde2ec6aaf54d5/config/initializers/github_client.rb#L19-L23

Comment (jeffrafter) [reply config/initializers/github_client.rb]: Confirmed this is a bug and great catch, thanks @favonia! I'll work on a regression test!

Comment (jeffrafter) [reply config/initializers/github_client.rb]: Actually in digging into this more, I don't expect it to regress and I think your fix is great.

**Review by jeffrafter — APPROVED** [review]
Thanks for the fix @favonia, your help has been extremely valuable in getting Classroom working on Enterprise!

**Pull request merged**